2024-07-11 12:54:44.468150: W tensorflow/stream_executor/platform/default/dso_loader.cc:55] Could not load dynamic library 'cudart64_100.dll'; dlerror: cudart64_100.dll not found
2024-07-11 12:54:44.468337: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
INFO flwr 2024-07-11 12:54:53,112 | app.py:165 | Starting Flower server, config: ServerConfig(num_rounds=10, round_timeout=None)
INFO flwr 2024-07-11 12:54:53,205 | app.py:179 | Flower ECE: gRPC server running (10 rounds), SSL is disabled
INFO flwr 2024-07-11 12:54:53,206 | server.py:89 | Initializing global parameters
INFO flwr 2024-07-11 12:54:53,206 | server.py:276 | Requesting initial parameters from one random client
INFO flwr 2024-07-11 12:54:54,478 | server.py:280 | Received initial parameters from one random client
INFO flwr 2024-07-11 12:54:54,479 | server.py:91 | Evaluating initial parameters
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:54:54,496 | server.py:98 | initial parameters (loss, other metrics): 0.6931471805599453, {'accuracy': 0.41}
INFO flwr 2024-07-11 12:54:54,497 | server.py:104 | FL starting
DEBUG flwr 2024-07-11 12:54:54,850 | server.py:227 | fit_round 1: strategy sampled 2 clients (out of 3)
DEBUG flwr 2024-07-11 12:54:54,878 | server.py:241 | fit_round 1 received 2 results and 0 failures
WARNING flwr 2024-07-11 12:54:54,880 | fedavgm.py:198 | No fit_metrics_aggregation_fn provided
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:54:54,889 | server.py:131 | fit progress: (1, 0.440925176869747, {'accuracy': 0.7866666666666666}, 0.3914010000000001)
DEBUG flwr 2024-07-11 12:54:54,889 | server.py:178 | evaluate_round 1: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:54:54,914 | server.py:192 | evaluate_round 1 received 5 results and 0 failures
WARNING flwr 2024-07-11 12:54:54,915 | fedavg.py:273 | No evaluate_metrics_aggregation_fn provided
DEBUG flwr 2024-07-11 12:54:54,915 | server.py:227 | fit_round 2: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:54:54,949 | server.py:241 | fit_round 2 received 5 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:54:54,961 | server.py:131 | fit progress: (2, 0.4387239041553003, {'accuracy': 0.7866666666666666}, 0.46428519999999907)
DEBUG flwr 2024-07-11 12:54:54,961 | server.py:178 | evaluate_round 2: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:54:54,985 | server.py:192 | evaluate_round 2 received 5 results and 0 failures
DEBUG flwr 2024-07-11 12:54:54,985 | server.py:227 | fit_round 3: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:54:55,021 | server.py:241 | fit_round 3 received 5 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:54:55,032 | server.py:131 | fit progress: (3, 0.4387250099251634, {'accuracy': 0.7866666666666666}, 0.5348658000000004)
DEBUG flwr 2024-07-11 12:54:55,032 | server.py:178 | evaluate_round 3: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:54:55,057 | server.py:192 | evaluate_round 3 received 5 results and 0 failures
DEBUG flwr 2024-07-11 12:54:55,058 | server.py:227 | fit_round 4: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:54:55,095 | server.py:241 | fit_round 4 received 5 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:54:55,106 | server.py:131 | fit progress: (4, 0.4387250014810911, {'accuracy': 0.7866666666666666}, 0.609002799999999)
DEBUG flwr 2024-07-11 12:54:55,107 | server.py:178 | evaluate_round 4: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:54:55,135 | server.py:192 | evaluate_round 4 received 5 results and 0 failures
DEBUG flwr 2024-07-11 12:54:55,135 | server.py:227 | fit_round 5: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:54:55,181 | server.py:241 | fit_round 5 received 5 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:54:55,192 | server.py:131 | fit progress: (5, 0.43872500143038756, {'accuracy': 0.7866666666666666}, 0.6952236000000003)
DEBUG flwr 2024-07-11 12:54:55,192 | server.py:178 | evaluate_round 5: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:54:55,222 | server.py:192 | evaluate_round 5 received 5 results and 0 failures
DEBUG flwr 2024-07-11 12:54:55,223 | server.py:227 | fit_round 6: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:54:55,261 | server.py:241 | fit_round 6 received 5 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:54:55,273 | server.py:131 | fit progress: (6, 0.43872500142958176, {'accuracy': 0.7866666666666666}, 0.7756204999999987)
DEBUG flwr 2024-07-11 12:54:55,273 | server.py:178 | evaluate_round 6: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:54:55,299 | server.py:192 | evaluate_round 6 received 5 results and 0 failures
DEBUG flwr 2024-07-11 12:54:55,300 | server.py:227 | fit_round 7: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:54:55,334 | server.py:241 | fit_round 7 received 5 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:54:55,347 | server.py:131 | fit progress: (7, 0.438725001429582, {'accuracy': 0.7866666666666666}, 0.8496419)
DEBUG flwr 2024-07-11 12:54:55,347 | server.py:178 | evaluate_round 7: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:54:55,378 | server.py:192 | evaluate_round 7 received 5 results and 0 failures
DEBUG flwr 2024-07-11 12:54:55,378 | server.py:227 | fit_round 8: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:54:55,416 | server.py:241 | fit_round 8 received 5 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:54:55,429 | server.py:131 | fit progress: (8, 0.4387250014295821, {'accuracy': 0.7866666666666666}, 0.9323211999999987)
DEBUG flwr 2024-07-11 12:54:55,429 | server.py:178 | evaluate_round 8: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:54:55,465 | server.py:192 | evaluate_round 8 received 6 results and 0 failures
DEBUG flwr 2024-07-11 12:54:55,466 | server.py:227 | fit_round 9: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:54:55,515 | server.py:241 | fit_round 9 received 6 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:54:55,528 | server.py:131 | fit progress: (9, 0.4356741606558022, {'accuracy': 0.78}, 1.0306625)
DEBUG flwr 2024-07-11 12:54:55,528 | server.py:178 | evaluate_round 9: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:54:55,569 | server.py:192 | evaluate_round 9 received 6 results and 0 failures
DEBUG flwr 2024-07-11 12:54:55,569 | server.py:227 | fit_round 10: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:54:55,615 | server.py:241 | fit_round 10 received 6 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:54:55,627 | server.py:131 | fit progress: (10, 0.4356738689103123, {'accuracy': 0.78}, 1.1307634999999987)
DEBUG flwr 2024-07-11 12:54:55,628 | server.py:178 | evaluate_round 10: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:54:55,661 | server.py:192 | evaluate_round 10 received 6 results and 0 failures
INFO flwr 2024-07-11 12:54:55,661 | server.py:153 | FL finished in 1.1641455999999994
INFO flwr 2024-07-11 12:54:55,661 | app.py:225 | app_fit: losses_distributed [(1, 0.40152427057425183), (2, 0.39948850657258717), (3, 0.3994887527965364), (4, 0.39948874499116627), (5, 0.39948874499116627), (6, 0.39948874499116627), (7, 0.39948874499116627), (8, 0.41340219846793586), (9, 0.41293945099626267), (10, 0.41293934030192236)]
INFO flwr 2024-07-11 12:54:55,661 | app.py:226 | app_fit: metrics_distributed_fit {}
INFO flwr 2024-07-11 12:54:55,661 | app.py:227 | app_fit: metrics_distributed {}
INFO flwr 2024-07-11 12:54:55,662 | app.py:228 | app_fit: losses_centralized [(0, 0.6931471805599453), (1, 0.440925176869747), (2, 0.4387239041553003), (3, 0.4387250099251634), (4, 0.4387250014810911), (5, 0.43872500143038756), (6, 0.43872500142958176), (7, 0.438725001429582), (8, 0.4387250014295821), (9, 0.4356741606558022), (10, 0.4356738689103123)]
INFO flwr 2024-07-11 12:54:55,662 | app.py:229 | app_fit: metrics_centralized {'accuracy': [(0, 0.41), (1, 0.7866666666666666), (2, 0.7866666666666666), (3, 0.7866666666666666), (4, 0.7866666666666666), (5, 0.7866666666666666), (6, 0.7866666666666666), (7, 0.7866666666666666), (8, 0.7866666666666666), (9, 0.78), (10, 0.78)]}
>>>>>>>>>>>>> setting initial parameters
[[0. 0. 0. 0. 0. 0. 0. 0.]
 [0. 0. 0. 0. 0. 0. 0. 0.]]
[array([[0., 0., 0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0., 0., 0.]]), array([0., 0.])]
[array([[-2.02819535,  1.0003751 ,  0.82501325,  0.38046936,  0.29285127,
         0.15790498,  0.02459004,  0.05981202]]), array([-0.02444476])]
[array([[-2.00803843,  1.04512883,  0.80589519,  0.42410503,  0.21796989,
         0.09146834,  0.03354162, -0.00469629]]), array([-0.0311493])]
[array([[-2.00801016,  1.04499132,  0.80589689,  0.42409303,  0.21805034,
         0.09145464,  0.03352138, -0.00468161]]), array([-0.03105647])]
[array([[-2.00801   ,  1.04499109,  0.80589775,  0.42409269,  0.21805101,
         0.09145333,  0.03352085, -0.00468119]]), array([-0.03105633])]
[array([[-2.00801   ,  1.04499109,  0.80589775,  0.42409269,  0.21805101,
         0.09145333,  0.03352085, -0.00468118]]), array([-0.03105633])]
[array([[-2.00801   ,  1.04499109,  0.80589775,  0.42409269,  0.21805101,
         0.09145333,  0.03352085, -0.00468118]]), array([-0.03105633])]
[array([[-2.00801   ,  1.04499109,  0.80589775,  0.42409269,  0.21805101,
         0.09145333,  0.03352085, -0.00468118]]), array([-0.03105633])]
[array([[-2.00801   ,  1.04499109,  0.80589775,  0.42409269,  0.21805101,
         0.09145333,  0.03352085, -0.00468118]]), array([-0.03105633])]
[array([[-1.93538119,  1.00273562,  0.7282539 ,  0.37076055,  0.21408658,
         0.08523224,  0.05574357, -0.00743508]]), array([0.03067338])]
[array([[-1.93537843,  1.0028046 ,  0.72833179,  0.37073603,  0.21405108,
         0.08517461,  0.05572125, -0.0074383 ]]), array([0.03066051])]
>>>>>>>>>>>>>run time:
0.484375s
